### ISSUE TITLE
Can't delete a specific document in V4.1.4

### ISSUE THREAD

**Issue by chucksullivancvs** [mbdavid/LiteDB#1820]
I have a collection with several documents I can't delete using Collection.Delete whether I delete by bson id or query.

For these certain documents I get KeyNotFoundException errors.

I also tried deleting these documents from LiteDB.Shell.exe and LiteDB Viewer, get the same error and it crashes the tools.

These same documents are returned in queries and otherwise behave like normal object documents.

I can somewhat work around it but would really like to delete them.

I'm looking for a recommendation on a quick work around.

I suppose I could exclude select these documents drop and copy the good docs to a new instance of the collection. But would not want to have to do that all the time or codify it. 

Note: I just downgraded to V4 to eliminate file locking issues from a network file share in V5. In V5 after upgrading the db file from V4, I had no problems deleting these same documents using DeleteMany in V5.

**Comment by lbnascimento** [comment]
@chucksullivancvs You could try running `db.Shrink()`. Also, I would like for you to send me this data file if possible (if it's not too big and doesn't contain sensitive information, of course).

**Comment by chucksullivancvs** [comment]
@lbnascimento - thx Leonardo. That worked. I did a db.Shrink first from LiteDB.Shell.exe and that fixed the KeyNotFoundException because I was then able to delete the items. I added a db.Shrink to my delete method so hopefully that will mitigate the issue going forward. 

As usual, thanks for the quick and meaningful response. LiteDB is a great, simple to use component for when a client-server db is not practical. It is serving me well. Keep up the good work guys, it is much appreciated.
